src/engine/square.ts

```diff
@@ -18,4 +24,8 @@ export default class Square {
     public toString() {
         return `Row ${this.row}, Col ${this.col}`;
     }
+
+    public moveBy(rowChange: number, colChange: number) {
+        return new Square(this.row + rowChange, this.col + colChange)
+    }
 }
```

Comment:
can these two functions not be one? And we would poss in `direction.rowChange` or `rowChange` etc